es-design-system/pages/organisms/es-file-thumbnail.vue

```diff
@@ -0,0 +1,221 @@
+<template>
+    <div>
+        <h1>
+            EsFileThumbnail
+        </h1>
+        <div class="my-5">
+            <h2>
+                Thumbnail for an image upload
+            </h2>
+            <p>
+                Click to see EsPreviewModal
+            </p>
+            <es-file-thumbnail
+                file-name="bill.jpg"
+                file-size="1.3 MB"
+                mime-type="image/jpg"
+                file-source="https://www.eversource.com/content/images/default-source/bills/ct-sample-bill-electric-front.jpg?sfvrsn=495ad262_4"
+                :percent-loaded="100"
+                @removeFile="removeFileAlert"
+                @showPreview="showPreviewModal=true" />
+            <es-file-preview-modal
+                file-source="https://www.eversource.com/content/images/default-source/bills/ct-sample-bill-electric-back.jpg?sfvrsn=e75bd262_4"
+                file-name="bill.png"
+                :visible="showPreviewModal"
+                @hidden="showPreviewModal=false" />
+        </div>
+        <h2>
+            Thumbnails for file uploads
+        </h2>
+        <div class="d-md-flex flex-nowrap mb-5">
+            <div
+                v-for="file in thumbnails"
+                :key="file.name"
+                class="mt-3 mt-md-0 mr-md-4">
+                <es-file-thumbnail
+                    :file-name="file.name"
+                    file-size="2.4 MB"
+                    :file-source="file.source"
+                    :mime-type="file.mimeType"
+                    :percent-loaded="100"
+                    @removeFile="removeFileAlert" />
+            </div>
+        </div>
+        <h2>
+            Thumbnail in loading state
+        </h2>
+        <div class="mb-3">
+            <es-file-thumbnail
+                file-name="bill.jpg"
+                mime-type="image/jpg"
+                :percent-loaded="percentLoaded"
+                :file-source="fileSource"
+                @removeFile="removeFileAlert"
+                @showPreview="showPreviewModal=true" />
+        </div>
+        <es-button @click="showLoadingState">
+            Show loading state
+        </es-button>
```

Comment:
I can click this a bunch of times and cause odd behavior. Can we disable it while its loading to 100%?